core/src/main/java/org/ehcache/core/EhcacheManager.java

```diff
@@ -111,76 +61,26 @@ public EhcacheManager(Configuration config) {
-    this.cacheManagerClassLoader = config.getClassLoader() != null ? config.getClassLoader() : ClassLoading.getDefaultClassLoader();
-    this.configuration = new DefaultConfiguration(config);
-    validateServicesConfigs();
-  }
 
-
-  private void validateServicesConfigs() {
-    HashSet<Class> classes = new HashSet<Class>();
-    for (ServiceCreationConfiguration<?> service : configuration.getServiceCreationConfigurations()) {
-      if (!classes.add(service.getServiceType())) {
-        throw new IllegalStateException("Duplicate creation configuration for service " + service.getServiceType());
-      }
-    }
+  public EhcacheManager(Configuration config, Collection<Service> services, boolean useLoaderInAtomics) {
+    super(config, services, useLoaderInAtomics);
   }
 
-
   @Override
-  public <K, V> Cache<K, V> getCache(String alias, Class<K> keyType, Class<V> valueType) {
-    statusTransitioner.checkAvailable();
-    final CacheHolder cacheHolder = caches.get(alias);
-    if(cacheHolder == null) {
-      return null;
-    } else {
-      try {
-        return cacheHolder.retrieve(keyType, valueType);
-      } catch (IllegalArgumentException e) {
-        throw new IllegalArgumentException("Cache '" + alias + "' type is <" + cacheHolder.keyType.getName() + ", "
-                                           + cacheHolder.valueType.getName() + ">, but you retrieved it with <"
-                                           + keyType.getName() + ", " + valueType.getName() +">");
-      }
-    }
+  protected Class<?> getServiceDependencyMarkerClass() {
+    return EhcacheManager.ServiceDeps.class;
   }
 
   @Override
-  public void removeCache(final String alias) {
-    removeCache(alias, true);
-  }
-
-  private void removeCache(final String alias, final boolean removeFromConfig) {
-    statusTransitioner.checkAvailable();
-    final CacheHolder cacheHolder = caches.remove(alias);
-    if(cacheHolder != null) {
-      final Ehcache<?, ?> ehcache = cacheHolder.retrieve(cacheHolder.keyType, cacheHolder.valueType);
-      if(!statusTransitioner.isTransitioning()) {
-        for (CacheManagerListener listener : listeners) {
-          listener.cacheRemoved(alias, ehcache);
-        }
-      }
-      closeEhcache(alias, ehcache);
-      if (removeFromConfig) {
-        configuration.removeCacheConfiguration(alias);
-      }
-      LOGGER.info("Cache '{}' is removed from EhcacheManager.", alias);
-    }
-  }
-
-  void closeEhcache(final String alias, final Ehcache<?, ?> ehcache) {
+  protected void closeEhcache(final String alias, final Ehcache<?, ?> ehcache) {
     boolean diskTransient = isDiskTransient(ehcache);
-    ehcache.close();
     if (diskTransient) {
       try {
         destroyPersistenceSpace(alias);
       } catch (CachePersistenceException e) {
-        LOGGER.debug("Unable to clear persistence space for cache {}", alias, e);
+        this.getLogger().debug("Unable to clear persistence space for cache {}", alias, e);
```

Comment:
I agree with @lorban on this one ... something non-fatal happened that might deserve attention by the user.  We're logging at an info level cache creation, cache closure, and (normal) cache removal -- so this can't be a concern about noise.

This is pre-existing code but I'm going to change this to warn unless someone has a strong objection.